### DISCUSSION TITLE
Website tutorial error 'Can't resolve 'fs''

### DISCUSSION THREAD

**Discussion by SaroGFX** [vercel/next.js#41449] [Help] [answered]
Hi!

I am trying to do the website tutorial but I'm getting stuck here: https://nextjs.org/learn/basics/data-fetching/implement-getstaticprops

I am getting this error 'Can't resolve 'fs' 

![Screenshot 2022-10-15 at 22 43 57](https://user-images.githubusercontent.com/3781800/196006843-690bb911-ce1d-4a49-971b-2c601b25289e.png)

I tried to redo all the steps but I get the same result. Do you guys also get the same result, is this a problem on the tutorial itself or do I have some local configuration that's different somehow? 

**I created a reproduction repo:** https://github.com/SaroGFX/nextjs-tutorial-fs-error

Thank you

**Comment by icyJoseph** [comment, marked as answer]
Here: https://github.com/SaroGFX/nextjs-tutorial-fs-error/blob/main/pages/index.js#L20, you are invoking a function that contains a `Node` module (fs) on the browser, when this functions is rehydrated by React.

```tsx
 <Layout home>
      {console.log(getSortedPostsData())}
      <Head>
```

Remove that log.

When in doubt, you could've defined `getSortedPostsData` on the same file and run it through https://next-code-elimination.vercel.app/, which outputs what code will be included on the client. You could also just do this with the current setup, but I think you are not so clear on what `getSortedPostData` implies, and how SSR/CSR works in Next.js, so take it one step at a time.

**Reply by SaroGFX** [reply]
Thanks for pointing that out, I understand now why that is an issue. The reason why I added that log is because I got this 'undefined' error and I was trying to debug it. 

If you remove the log, you get this issue: 

![Screenshot 2022-10-17 at 10 12 48](https://user-images.githubusercontent.com/3781800/196124141-e44be15f-3c68-4945-a55d-58f55165bd3f.png)

Any ideas what is the problem here?

**Reply by icyJoseph** [reply]
Yeah, in `_app` you have a typo, it should be `pageProps`, but you have `PageProps`:

```js
import "../styles/global.css";
 
// Here 👀
const App = ({ Component, pageProps }) => {
  return <Component {...pageProps} />;
};

export default App;
```

**Reply by SaroGFX** [reply]
Aha that's it, thank you very much!
